### PR TITLE
Pandas FutureWarnings are disrupting tqdm progress bars

### LINKED ISSUE #793: Pandas FutureWarnings are disrupting tqdm progress bars
### Environment Details

When calling some functions in SDV like `synthesizer.sample_remaining_columns()`, the tqdm progress bar doesn't progress in-line. Instead, each update to the progress bar results in the display of a `FutureWarning` from pandas around the soon-to-be-deprecated `fillna()` function.

https://github.com/sdv-dev/RDT/assets/801507/436f787e-d09e-42f5-9b52-2b943949bb55

* RDT version: 1.11.0

### Proposed Solution

After a quick discussion with Neha, we should update the code to move away from these functions (not just suppress the warnings). We only use this function in a few places (4) in the RDT codebase:

https://github.com/search?q=repo%3Asdv-dev%2FRDT%20fillna&type=code

### PR DESCRIPTION
**Pull request by frances-h**
CU-86b01yxhv
Resolve #793

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b01yxhv RDT - Pandas FutureWarnings are disrupting tqdm progress bars #793](https://app.clickup.com/t/86b01yxhv)

**Review by fealho — COMMENTED** [review]
Is it not possible to do `df.infer_objects(copy=False).fillna(value)`, as is suggested [here](https://stackoverflow.com/questions/77900971/pandas-futurewarning-downcasting-object-dtype-arrays-on-fillna-ffill-bfill)?

File: rdt/transformers/null.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """Transformer for data that contains Null values."""
 
 import logging
+import warnings
```
Comment (gsheni) [inline rdt/transformers/null.py:4]: Not needed now?

File: tests/integration/transformers/test_categorical.py [hunk 2]
```diff
@@ -198,6 +198,28 @@ def test_frequency_encoder_numerical_nans():
...
+    column = 'column_name'
+
+    # Run and Assert
+    transformer = FrequencyEncoder()
+    with warnings.catch_warnings():
+        warnings.simplefilter('error', FutureWarning)
+        transformer.fit(data, column)
+        transformed = transformer.transform(data)
+        reverse = transformer.reverse_transform(transformed)
+
```
Comment (gsheni) [inline tests/integration/transformers/test_categorical.py:219]:
```suggestion
    with pytest.warns(record=True) as record:
        transformer.fit(data, column)
        transformed = transformer.transform(data)
        reverse = transformer.reverse_transform(transformed)
    # look in records to make sure no FutureWarning raised?
```

Comment (gsheni) [reply tests/integration/transformers/test_categorical.py:219]: Oh, I see. Nvm you did it right.
https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests

**Review by fealho — APPROVED** [review]
Looks good, I'd just update the title of the PR.

**Review by rwedge — APPROVED** [review]

**Pull request merged**